egs/wsj/s5/steps/nnet3/nnet3_train_lib.py

```diff
@@ -506,41 +639,53 @@ def DoShrinkage(iter, model_file, non_linearity, shrink_threshold):
 
     return False
 
-def ComputeTrainCvProbabilities(dir, iter, egs_dir, run_opts, mb_size=256, wait = False):
+def ComputeTrainCvProbabilities(dir, iter, egs_dir, run_opts, mb_size=256, wait = False, use_raw_nnet = False, compute_accuracy = True):
 
-    model = '{0}/{1}.mdl'.format(dir, iter)
+    if use_raw_nnet:
+        model = "{dir}/{iter}.raw".format(dir = dir, iter = iter)
+    else:
+        model = "nnet3-am-copy --raw=true {dir}/{iter}.mdl - |".format(dir = dir, iter = iter)
+
+    compute_prob_opts = ""
+    if compute_accuracy:
+        compute_prob_opts = "--compute-accuracy"
 
     RunKaldiCommand("""
 {command} {dir}/log/compute_prob_valid.{iter}.log \
-  nnet3-compute-prob "nnet3-am-copy --raw=true {model} - |" \
+  nnet3-compute-prob {compute_prob_opts} "{model}" \
         "ark,bg:nnet3-merge-egs --minibatch-size={mb_size} ark:{egs_dir}/valid_diagnostic.egs ark:- |"
     """.format(command = run_opts.command,
                dir = dir,
                iter = iter,
                mb_size = mb_size,
                model = model,
+               compute_prob_opts = compute_prob_opts,
                egs_dir = egs_dir), wait = wait)
 
     RunKaldiCommand("""
 {command} {dir}/log/compute_prob_train.{iter}.log \
-  nnet3-compute-prob "nnet3-am-copy --raw=true {model} - |" \
+  nnet3-compute-prob {compute_prob_opts} "{model}" \
        "ark,bg:nnet3-merge-egs --minibatch-size={mb_size} ark:{egs_dir}/train_diagnostic.egs ark:- |"
     """.format(command = run_opts.command,
                dir = dir,
                iter = iter,
                mb_size = mb_size,
                model = model,
+               compute_prob_opts = compute_prob_opts,
                egs_dir = egs_dir), wait = wait)
 
+def ComputeProgress(dir, iter, egs_dir, run_opts, mb_size=256, wait=False, use_raw_nnet = False):
```

Comment:
`use_raw_nnet` does not seem to be appropriate name for the variable as you are using raw networks in both cases. Better use the name `get_raw_nnet_from_am=True`.